actors/core/src/main/java/com/ea/orbit/actors/ObserverManager.java

```diff
@@ -129,22 +144,26 @@ public boolean addObserver(final T observer)
      */
     public void notifyObservers(final Consumer<T> callable)
     {
-        final List<T> fail = new ArrayList<>(0);
-        observers.forEach(o -> {
+        List<T> fail = null;
+        for (T observer : observers)
+        {
             try
             {
-                callable.accept(o);
+                callable.accept(observer);
             }
             catch (final Exception ex)
             {
-                fail.add(o);
+                if (fail == null) {
```

Comment:
I considered invoking remove directly here but looking at the source for remove and removeAll I suspect the later takes less CPU time with large collections.